homeassistant/components/sensor/rmvtransport.py

```diff
@@ -0,0 +1,209 @@
+    """Set up the RMV departure sensor."""
+    sensors = []
+    for nextdeparture in config.get(CONF_NEXT_DEPARTURE):
+        sensors.append(
+            RMVDepartureSensor(
+                nextdeparture.get(CONF_STATION),
+                nextdeparture.get(CONF_DESTINATIONS),
+                nextdeparture.get(CONF_DIRECTIONS),
+                nextdeparture.get(CONF_LINES),
+                nextdeparture.get(CONF_PRODUCTS),
+                nextdeparture.get(CONF_TIMEOFFSET),
+                nextdeparture.get(CONF_MAXJOURNEYS),
+                nextdeparture.get(CONF_NAME)))
+    add_entities(sensors, True)
+
+
+class RMVDepartureSensor(Entity):
+    """Implementation of an RMV departure sensor."""
+
+    def __init__(self, station, destinations, directions,
+                 lines, products, timeoffset, maxjourneys, name):
+        """Initialize the sensor."""
+        self._station = station
+        self._name = name
+        self.data = RMVDepartureData(station, destinations, directions,
+                                     lines, products, timeoffset, maxjourneys)
+        self._state = STATE_UNKNOWN
+        self._icon = ICONS['-']
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        if self._state is STATE_UNKNOWN:
+            return False
+        return True
+
+    @property
+    def state(self):
+        """Return the next departure time."""
+        return self._state
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        result = {}
+        try:
+            result = {
+                'next_departures': [val for val in self.data.departures[1:]],
+                'direction': self.data.departures[0].get('direction', '-'),
+                'line': self.data.departures[0].get('number', '-'),
+                'minutes': self.data.departures[0].get('minutes', '-'),
+                'product': self.data.departures[0].get('product', '-'),
+            }
+        except IndexError:
+            pass
+        return result
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit this state is expressed in."""
+        return "min"
+
+    def update(self):
+        """Get the latest data and update the state."""
+        self.data.update()
+        if not self.data.departures:
+            self._state = '-'
+            self._icon = ICONS['-']
+        else:
+            if self._name is DEFAULT_NAME:
+                self._name = self.data.station
+            else:
+                self._name = self._name
+            self._station = self.data.station
+            self._state = self.data.departures[0].get('minutes', '-')
```

Comment:
We don't want to use relative time as state, please use datetime instead. If not exist, None should be use, not '-'.

In future, we will have timestamp device class to display datetime as relative time in frontend, see https://github.com/home-assistant/architecture/issues/39